File: C++/TwoSum.cpp
#include <unordered_map>
#include <algorithm>
#include<vector>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> map{};
        int count{0};
        for(auto num : nums){
            ++count;
            if(map.find(num) != map.end()){
                vector<int> sol{count,map[num]};
                std::sort( sol.begin(), sol.end() );
                return sol;
            }
            map.insert( {target-num, count});
        }
    }
};

File: Go/Two Sum.go
func twoSum(nums []int, target int) []int {
    x := make([]int, 0, 2)
    m := make(map[int]int)
    
    for i, num := range nums {
        j, e := m[target-num]
        if e {
            x = append(x, i)
            x = append(x, j)
            break
        }
        m[num] = i
    }

    return x
}
